drivers/hmis_supplemental/app/controllers/hmis_supplemental/client_data_sets_controller.rb

```diff
@@ -58,9 +58,10 @@ def load_authorized_data_set
     end
 
     def client_groups
+      # note: order is important here, source_visible_to appears to clobber the data source condition
```

Comment:
If `visible_by` clobbers the data source id, don't we need to check that `@data_set.data_source_id` is in `visible_by.select(:data_source_id)`?  Is there a possibility you could get here without having that check performed on `@data_set` first?

Reply:
if you use `where(c_t[:data_source_id].eq(@data_set.data_source_id))` does the clobber still happen?  I think we use the arel versions pretty heavily in the `source_visible_to` scope, so maybe there's an odd interaction with the active record version. (Not sure that's a better solution, but it might mean the order isn't important).

Reply:
> if you use where(c_t[:data_source_id].eq(@data_set.data_source_id)) does the clobber still happen? 

Good idea but unfortunately I do see the same order-dependent behavior using arel. Glass-half full is that AR is consistent at least.

> If source_visible_to clobbers the data source id, don't we need to check that @data_set.data_source_id is in source_visible_to(current_user).select(:data_source_id)?

Good point, we do check that data set is visible to user but not the data source. It does seem possible that a user could see the data set but not the data source. I guess the data set permission doesn't imply permission to see the client data?

I'll update `DataSet.viewable_by` to reflect this assumption; checking for `can_view_supplemental_client_data for` on the data source as well. So then the permission to see the data set tabs are:
- user must be using acls
- user needs `can_view_supplemental_client_data` on a collection that includes the data set
- user needs `can_view_supplemental_client_data` on the associated data_source as well

Reply:
Let's give that a go and see if the data source is also necessary long-term.  Feels safer.